src/image/loading_displaying.js

```diff
@@ -158,6 +159,341 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
+    /*
+      we draw the next frame. this is important, since
+      busy sketches or low end devices might take longer
+      to render some frames. So we just wait for the frame
+      to be drawn and immediately save it to a buffer and continue
+      */
+    this.redraw();
+
+    // depending on the context we'll extract the pixels one way
+    // or another
+    let data = undefined;
+
+    if (this.drawingContext instanceof WebGLRenderingContext) {
+      pixels = new Uint8Array(
+        gl.drawingBufferWidth * gl.drawingBufferHeight * 4
+      );
+      gl.readPixels(
+        0,
+        0,
+        gl.drawingBufferWidth,
+        gl.drawingBufferHeight,
+        gl.RGBA,
+        gl.UNSIGNED_BYTE,
+        pixels
+      );
+
+      data = _flipPixels(pixels);
+    } else {
+      data = this.drawingContext.getImageData(0, 0, this.width, this.height)
+        .data;
+    }
+
+    frames.push(data);
+    count++;
+
+    p.html(
+      'Saved frame <b>' +
+        frames.length.toString() +
+        '</b> out of ' +
+        nFrames.toString()
+    );
+    await new Promise(resolve => setTimeout(resolve, 0));
+  }
+  p.html('Frames processed, generating color palette...');
+
+  this.loop();
+  this.pixelDensity(lastPixelDensity);
+
+  // create the gif encoder and the colorspace format
+  const gif = GIFEncoder();
+
+  // calculate the global palette for this set of frames
+  const globalPalette = _generateGlobalPalette(frames);
+
+  // the way we designed the palette means we always take the last index for transparency
+  const transparentIndex = globalPalette.length - 1;
+
+  // we are going to iterate the frames in pairs, n-1 and n
+  for (let i = 0; i < frames.length; i++) {
+    if (i === 0) {
+      const indexedFrame = applyPalette(frames[i], globalPalette, {
+        format: 'rgba4444'
+      });
+      gif.writeFrame(indexedFrame, this.width, this.height, {
+        palette: globalPalette,
+        delay: gifFrameDelay,
+        dispose: 1
+      });
+      continue;
+    }
+
+    // matching pixels between frames can be set to full transparency,
+    // kinda digging a "hole" into the frame to see the pixels that where behind it
+    // (which would be the exact same, so not noticeable changes)
+    // this helps make the file quite smaller
+    let currFramePixels = frames[i];
+    let lastFramePixels = frames[i - 1];
+    let matchingPixelsInFrames = [];
+    for (let p = 0; p < currFramePixels.length; p += 4) {
+      let currPixel = [
+        currFramePixels[p],
+        currFramePixels[p + 1],
+        currFramePixels[p + 2],
+        currFramePixels[p + 3]
+      ];
+      let lastPixel = [
+        lastFramePixels[p],
+        lastFramePixels[p + 1],
+        lastFramePixels[p + 2],
+        lastFramePixels[p + 3]
+      ];
+
+      // if the pixels are equal, save this index to be used later
+      if (_pixelEquals(currPixel, lastPixel)) {
+        matchingPixelsInFrames.push(parseInt(p / 4));
```

Comment:
Since `p` is already a number, I think you can use `Math.floor()` instead of `parseInt()`. Although since `p` is iterating by 4, I think maybe we don't need to do anything extra to ensure it's an integer here?